src/Http/WebUtilities/test/MultipartSectionPipeReaderTests.cs

```diff
@@ -0,0 +1,197 @@
+        }
+
+        private static string GetString(ReadOnlySequence<byte> buffer)
+        {
+            return Encoding.ASCII.GetString(buffer);
+        }
+
+        [Theory]
+        [InlineData(TextAndBoundary, Text)]
+        [InlineData(HtmlWithNewLinesAndBoundary, HtmlWithNewLines)]
+        [InlineData(TextWithPartialBoundaryMatchAndBoundary, TextWithPartialBoundaryMatch)]
+        public async Task MultipartSectionPipeReader_ValidBody_Success(string input, string expected)
+        {
+            var pipeReader = MakeReader(input);
+            var sectionReader = new MultipartSectionPipeReader(pipeReader, new MultipartBoundary(Boundary));
+
+            var result = await sectionReader.ReadAsync();
+            Assert.False(result.IsCompleted);
+            Assert.False(result.IsCanceled);
+            Assert.False(result.Buffer.IsEmpty);
+
+            var actual = GetString(result.Buffer);
+            Assert.Equal(expected, actual);
+
+            sectionReader.AdvanceTo(result.Buffer.End);
+            result = await sectionReader.ReadAsync();
+            Assert.True(result.IsCompleted);
+            Assert.False(result.IsCanceled);
+            Assert.True(result.Buffer.IsEmpty);
+        }
+
+        [Theory]
+        [InlineData(TextAndBoundary, Text, 1)]
+        [InlineData(HtmlWithNewLinesAndBoundary, HtmlWithNewLines, 4)]
+        [InlineData(TextWithPartialBoundaryMatchAndBoundary, TextWithPartialBoundaryMatch, 2)]
+        public async Task MultipartSectionPipeReader_SingleByteReadAdvance_Success(string input, string expected, int reads)
+        {
+            var pipeReader = MakeSingleByteReader(input);
+            var sectionReader = new MultipartSectionPipeReader(pipeReader, new MultipartBoundary(Boundary));
+            var bufferBuilder = new List<byte>();
+
+            for (int i = 0; i < reads; i++)
+            {
+                var temp = await sectionReader.ReadAsync();
+                Assert.False(temp.IsCompleted);
+                Assert.False(temp.IsCanceled);
+                Assert.False(temp.Buffer.IsEmpty);
+
+                bufferBuilder.AddRange(temp.Buffer.ToArray());
+
+                sectionReader.AdvanceTo(temp.Buffer.End);
+            }
+
+            var actual = GetString(new ReadOnlySequence<byte>(bufferBuilder.ToArray()));
+            Assert.Equal(expected, actual);
+
+            var result = await sectionReader.ReadAsync();
+            Assert.True(result.IsCompleted);
+            Assert.False(result.IsCanceled);
+            Assert.True(result.Buffer.IsEmpty);
+        }
+
+        [Theory]
+        [InlineData(TextAndBoundary, Text, 1)]
+        [InlineData(HtmlWithNewLinesAndBoundary, HtmlWithNewLines, 52)]
+        [InlineData(TextWithPartialBoundaryMatchAndBoundary, TextWithPartialBoundaryMatch, 14)]
+        public async Task MultipartSectionPipeReader_SingleByteReadWithoutAdvance_Success(string input, string expected, int reads)
```

Comment:
Not sure how it differs from  WithAdvance Test. The only difference is that the result isn't buffered and the expected bytes read is larger - both seem like a mistake in the test rather than the functionality, since I am not sure how it's even supposed to pass - **I am gonna guess that this is a copy pasted test that was never implemented, but I am double checking.**

Reply:
The difference is that the prior test always calls `AdvanceTo(temp.Buffer.End);` but this one calls `AdvanceTo(temp.Buffer.Start, temp.Buffer.End);`. That makes the former unbuffered and the latter buffered. Both are valid scenarios.

Reply:
Oh, I might have missed that. Anyways, I was wondering why they were expecting different number of reads, since they are both pretty similar. I am guessing that this was a mistake.